src/main/java/org/spdx/utility/compare/LicenseCompareHelper.java

```diff
@@ -843,81 +844,173 @@ private static int findOriginalStart(int position, List<Pair<Integer, Integer>>
-			String completeText = normalizedText.substring(startIndex, endIndex);
-			try {
-				result = !isTextStandardLicense(license, completeText).isDifferenceFound();
-			} catch (SpdxCompareException e) {
-				logger.warn("Compare exception for license ID "+license.getLicenseId(), e);
-				result = false;
-			} catch (InvalidSPDXAnalysisException e) {
-				logger.warn("Compare exception for license ID "+license.getLicenseId(), e);
-				result = false;
+
+		return result;
+	}
+
+
+	/**
+	 * Returns a list of SPDX Standard License ID's from the provided list that were found in the text, using
+	 * the SPDX matching guidelines.
+	 * @param text Text to compare to the standard license texts
+	 * @param licenseIds License ids to compare against
+	 * @return List of SPDX standard license IDs from licenseIds that match
+	 * @throws InvalidSPDXAnalysisException If an error occurs accessing the standard licenses
+	 * @throws SpdxCompareException If an error occurs in the comparison
+	 */
+	public static List<String> matchingStandardLicenseIds(String text, List<String> licenseIds) throws InvalidSPDXAnalysisException, SpdxCompareException {
+		List<String> result = new ArrayList<>();
+
+		if (text != null && !text.isEmpty() && licenseIds != null && !licenseIds.isEmpty()) {
+			for (String stdLicId : licenseIds) {
+				SpdxListedLicense license = ListedLicenses.getListedLicenses().getListedLicenseById(stdLicId);
+				if (isStandardLicenseWithinText(text, license)) {
+					result.add(license.getLicenseId());
+				}
 			}
 		}
+
 		return result;
 	}
 
-	
 	/**
-	 * Returns a list of SPDX Standard License ID's that match the text provided using
+	 * Returns a list of SPDX Standard License ID's that were found in the text, using
 	 * the SPDX matching guidelines.
-	 * @param licenseText Text to compare to the standard license texts
-	 * @return Array of SPDX standard license IDs that match
+	 * @param text Text to compare to the standard license texts
+	 * @return List of SPDX standard license IDs that match
 	 * @throws InvalidSPDXAnalysisException If an error occurs accessing the standard licenses
 	 * @throws SpdxCompareException If an error occurs in the comparison
 	 */
-	public static String[] matchingStandardLicenseIds(String licenseText) throws InvalidSPDXAnalysisException, SpdxCompareException {
-		List<String> stdLicenseIds = ListedLicenses.getListedLicenses().getSpdxListedLicenseIds();
-		List<String> matchingIds  = new ArrayList<>();
-		for (String stdLicId : stdLicenseIds) {
-			SpdxListedLicense license = ListedLicenses.getListedLicenses().getListedLicenseById(stdLicId);
-			if (!isTextStandardLicense(license, licenseText).isDifferenceFound()) {
-				matchingIds.add(license.getLicenseId());
+	public static List<String> matchingStandardLicenseIds(String text) throws InvalidSPDXAnalysisException, SpdxCompareException {
+		return matchingStandardLicenseIds(text, ListedLicenses.getListedLicenses().getSpdxListedLicenseIds());
+	}
```

Comment:
BTW - the same optimization mentioned above could be used for the existing code as well.

Reply:
I didn't make this optimisation to the existing code, given I clearly don't understand it well enough yet.  😉